### PR TITLE
Use real authentication my profile routes

### PR DESCRIPTION
**Pull request by Devessier**

### CODE REVIEW

File: packages/server/app/Controllers/Http/UserSettingsController.ts [hunk 1]
```diff
@@ -6,22 +6,26 @@ import {
...
+        const user = auth.user;
+        invariant(
+            user !== undefined,
+            'User must be logged in to get her settings',
+        );
 
-        const user = await User.findOrFail(tmpAuthUserID);
-        await user.load('playlistsVisibilitySetting');
-        await user.load('relationsVisibilitySetting');
+        await user.load((loader) => {
```
Comment (prastoin) [inline packages/server/app/Controllers/Http/UserSettingsController.ts:24]: praise: Indeed better ! I didn't knew this way

**Review by prastoin — DISMISSED** [review]
Great work !
Added few comments but nothing blocking the merge
Then ready to be merged !

File: packages/server/start/routes.ts [hunk 2]
```diff
@@ -45,6 +45,41 @@ Route.post(
     'MpeRoomsHttpController.listAllRooms',
 ).middleware('every-auth');
 
+export const MY_PROFILE_ROUTES_GROUP_PREFIX = '/me';
+Route.group(() => {
```
[the remark this answers is not in the document]
Comment (Devessier) [reply packages/server/start/routes.ts:49]: Right!

File: packages/client/screens/MySettings/__tests__/UpdateVisibilitySettings.test.tsx [hunk 3]
```diff
@@ -618,33 +618,32 @@ const updateVisibilitySettingsTestModel = createTestModel<TestingContext>(
...
-                Record<string, never>,
-                GetMySettingsResponseBody
-            >(`${SERVER_ENDPOINT}/me/settings`, (req, res, ctx) => {
-                const user = db.myProfileInformation.findFirst({
-                    where: {
-                        userID: {
-                            equals: req.body.tmpAuthUserID,
+            rest.get<never, never, GetMySettingsResponseBody>(
+                `${SERVER_ENDPOINT}/me/settings`,
+                withAuthentication((_req, res, ctx) => {
```
[the remark this answers is not in the document]
Comment (Devessier) [reply packages/client/screens/MySettings/__tests__/UpdateVisibilitySettings.test.tsx:623]: And nether did it to me.
In fact, what I dislike the most is that we evenly copy and paste a whole handler to add, for instance, a delay.
I know MSW has a feature called [Custom response composition](https://mswjs.io/docs/recipes/custom-response-composition) that could help with that, but I never used it.

**Review by prastoin — APPROVED** [review]
Ready to be merged !

**Pull request merged**